validator/src/command.rs

```diff
@@ -0,0 +1,226 @@
+use std::fmt::{Debug, Display};
+use std::fs;
+use std::str::FromStr;
+
+use anchor_client::solana_sdk::signature::{read_keypair_file, Keypair};
+use anchor_client::solana_sdk::signer::Signer;
+use clap::{arg, command, Args, Parser, Subcommand};
+use dialoguer::theme::ColorfulTheme;
+use dialoguer::Input;
+use log::LevelFilter;
+use serde::{Deserialize, Serialize};
+
+use crate::utils::{config_file, setup_logging};
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Config {
+    pub rpc_url: String,
+    pub ws_url: String,
+    pub program_id: String,
+    pub genesis_hash: String,
+    pub keypair: Vec<u8>,
+    pub log_level: String,
+}
+
+impl Display for Config {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let keypair = Keypair::from_bytes(&self.keypair).unwrap();
+        write!(
+            f,
+            "\nrpc_url: {}\nws_url: {}\nprogram_id: {}\ngenesis_hash: \
+             {}\nvalidator_public_key: {}\nlog_level: {}",
+            self.rpc_url,
+            self.ws_url,
+            self.program_id,
+            self.genesis_hash,
+            keypair.pubkey(),
+            self.log_level
+        )
```

Comment:
well i feel its better to display the public key. Displaying the complete key would be a security risk.

Reply:
Right, but that’s why I’m asking whether we want it to be shown in Debug implementation (which the current derive does).

Reply:
oh i see, well i dont think so.

Reply:
I mean if we are printing the pubkey only for the user to check if its the right keypair file or not. So i dont think printing the complete keypair would help.

Reply:
so i will write the Debug trait to print pubkey like how i did with format

Reply:
SG.  You’d probably wanna use [debug_struct](https://doc.rust-lang.org/std/fmt/struct.Formatter.html#method.debug_struct).

Reply:
sure sure will check that out.

Reply:
but why exactly would we need wrapper type?

Reply:
You could then define Debug and Display on the wrapper such that you’d be able to use Debug derive for this type.

Reply:
oh i see, so instead of implement `Display` and `Debug` for `Config` struct, just implementing it for wrapper type would work. Makes sense now.

Reply:
well since the `Config` struct is used to write to the toml config file, they need to implement `Serde`. We would have to write `Serde` traits for the wrapped type which doesnt seem to worth it to me. wdyt?